sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -193,106 +194,215 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
+
+    @property
+    def encoding(self) -> Optional[str]:
+        """Returns the response encoding.
+
+        :return: The response encoding. We either return the encoding set by the user,
+         or try extracting the encoding from the response's content type. If all fails,
+         we return `None`.
+        :rtype: optional[str]
+        """
+        ...
+
+    @encoding.setter
+    def encoding(self, value: Optional[str]) -> None:
+        """Sets the response encoding.
+
+        :rtype: None
+        """
 
     @property
-    def body(self):
-        return self.data
+    def url(self) -> str:
+        """The URL that resulted in this response.
 
-    @body.setter
-    def body(self, value):
-        self.data = value
+        :rtype: str
+        """
+        ...
 
-    def format_parameters(self, params):
-        """Format parameters into a valid query string.
-        It's assumed all parameters have already been quoted as
-        valid URL strings."""
+    @property
+    def content(self) -> bytes:
+        """Return the response's content in bytes.
 
-    def set_xml_body(self, data):
-        """Set an XML element tree as the body of the request."""
+        :rtype: bytes
+        """
+        ...
 
-    def set_json_body(self, data):
-        """Set a JSON-friendly object as the body of the request."""
+    def text(self, encoding: Optional[str] = None) -> str:
+        """Returns the response body as a string.
 
-    def set_multipart_body(self, data=None):
-        """Set form-encoded data as the body of the request.
-        Supported content-types are:
-            - application/x-www-form-urlencoded
-            - multipart/form-data
+        :param optional[str] encoding: The encoding you want to decode the text with. Can
+         also be set independently through our encoding property
+        :return: The response's content decoded as a string.
+        :rtype: str
         """
+        ...
 
-    def set_bytes_body(self, data):
-        """Set generic bytes as the body of the request."""
+    def json(self) -> Any:
+        """Returns the whole body as a json object.
 
-    def set_multipart_mixed(self, *requests, **kwargs):
-        """Set requests for a multipart/mixed body.
-        Optionally apply "policies" in kwargs to each request.
+        :return: The JSON deserialized response body
+        :rtype: any
+        :raises json.decoder.JSONDecodeError or ValueError (in python 2.7) if object is not JSON decodable:
         """
+        ...
+
+    def raise_for_status(self) -> None:
+        """Raises an HttpResponseError if the response has an error status code.
+
+        If response is good, does nothing.
+
+        :rtype: None
+        :raises ~azure.core.HttpResponseError if the object has an error status code.:
+        """
+        ...
+
+    def close(self) -> None:
+        ...
+
+    def read(self) -> bytes:
+        ...
+
+    def iter_raw(self, **kwargs: Any) -> Iterator[bytes]:
+        ...
```

Comment:
Is there any reason we only have introduction for async but not for sync?

Reply:
oh let me fix that